Kitodo-ImageManagementModule/src/main/java/org/kitodo/imagemanagementmodule/ImageManagementModule.java

```diff
@@ -0,0 +1,197 @@
+
+import java.awt.Image;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.net.URI;
+
+import javax.imageio.ImageIO;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.kitodo.api.imagemanagement.ImageFileFormat;
+import org.kitodo.api.imagemanagement.ImageManagementInterface;
+import org.kitodo.config.Config;
+
+/**
+ * An ImageManagementInterface implementation using ImageMagick.
+ */
+public class ImageManagementModule implements ImageManagementInterface {
+    private static final Logger logger = LogManager.getLogger(ImageManagementModule.class);
+
+    /**
+     * Image format used internally to create image derivatives, optimized for
+     * loss-free image quality, and speed. The format must be supported by both
+     * ImageMagick and {@link javax.imageio.ImageIO}.
+     */
+    private static final String RAW_IMAGE_FORMAT = ".bmp";
+
+    /**
+     * Temporary directory location.
+     */
+    private static final File TMPDIR = new File(
+            Config.getParameter("ImageManagementModule.tmpDir", System.getProperty("java.io.tmpdir")));
+
+    /**
+     * Image format used internally to create web images, optimized for small
+     * size. The format must be supported by both ImageMagick and
+     * {@link javax.imageio.ImageIO}.
+     */
+    private static final String WEB_IMAGE_FORMAT = ".jpeg";
+
+    /**
+     * {@inheritDoc}
+     *
+     * @see org.kitodo.api.imagemanagement.ImageManagementInterface#getScaledWebImage(java.net.URI,
+     *      double)
+     */
+    @Override
+    public Image getScaledWebImage(URI sourceUri, double factor) throws IOException {
+
+        if (!new File(sourceUri).exists()) {
+            throw new FileNotFoundException("sourceUri must exist: " + sourceUri.getRawPath());
+        }
+        if (Double.isNaN(factor)) {
+            throw new IllegalArgumentException("factor must be a number, but was " + Double.toString(factor));
+        }
+        if (factor <= 0.0) {
+            throw new IllegalArgumentException("factor must be > 0.0, but was " + Double.toString(factor));
+        }
+
+        File tempFile = File.createTempFile("scaledWebImage-", WEB_IMAGE_FORMAT, TMPDIR);
+        try {
+            tempFile.deleteOnExit();
+            ImageConverter imageConverter = new ImageConverter(sourceUri);
+            imageConverter.addResult(tempFile.toURI()).resize(factor);
+
+            logger.info("Generating scaled web image from {} as {}, factor {}%", sourceUri, tempFile, 100 * factor);
+            imageConverter.run();
+
+            logger.trace("Loading {}", tempFile);
+            Image buffer = ImageIO.read(tempFile);
+            logger.trace("{} successfully loaded", tempFile);
+            return buffer;
+        } finally {
+            logger.debug("Deleting {}", tempFile);
+            tempFile.delete();
+            logger.trace("Successfully deleted {}", tempFile);
```

Comment:
You are logging a successful deleting but you are interpreting the return value of `delete` call. You should adjust your code to use the return value for your logging statement.